app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/maps/GoogleMapsScreen.kt

```diff
@@ -1,49 +1,235 @@
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
+import com.github.leuludyha.domain.model.library.Mocks
+import com.github.leuludyha.domain.model.user.User
 import com.github.leuludyha.ibdb.R
+import com.github.leuludyha.ibdb.presentation.Orientation
+import com.github.leuludyha.ibdb.presentation.components.books.book_views.MiniBookView
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.Priority
+import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.model.CameraPosition
 import com.google.android.gms.maps.model.LatLng
 import com.google.maps.android.compose.*
+import kotlinx.coroutines.launch
 
 /**
- * For now, this screen only starts google maps and displays a few hardcoded markers at EPFL.
+ * This screen opens a google maps composable. Currently implemented features include user
+ * localisation and nearby users display. When clicking on a marker, it can open the actual
+ * google maps app to propose a route to get to the marker.
  */
 @Composable
 fun GoogleMapsScreen(
     navController: NavHostController,
-    initialLatitude: String?, //These are treated as strings because they are Double, but navController only accepts floats
-    initialLongitude: String?,
-    interestPoints: Array<String>?
+    paddingValues: PaddingValues,
+    viewModel: GoogleMapsScreenViewModel = hiltViewModel()
 ) {
-    //This below will be replaced by the current user's location later.
-    val epfl = LatLng(46.520536, 6.568318)
-
     val cameraPositionState = rememberCameraPositionState {
-        position = CameraPosition.fromLatLngZoom(epfl, 15f)
+        position = CameraPosition.fromLatLngZoom(viewModel.defaultLocation, 15f)
     }
 
+    val uiSettings by remember {
+        mutableStateOf(
+            MapUiSettings(
+                //I did my own implementation of this, I encountered problems and no documentation online
+                myLocationButtonEnabled = false,
+                zoomControlsEnabled = false
+            )
+        )
+    }
+    val context = LocalContext.current
+    val nearbyUsers by viewModel.nearbyUsers
+
     GoogleMap(
         modifier = Modifier
+            .padding(paddingValues)
             .fillMaxSize()
             .testTag("GoogleMaps::main"),
+        uiSettings = uiSettings,
         cameraPositionState = cameraPositionState
     ) {
         EpflMarkerAndLimits()
         SatelliteMarker()
```

Comment:
Will this marker (`SatelliteMarker`) change? I think it is a bit weird to have a toast displaying the coordinates, it isn't very useful for the user.

Reply:
This marker (and EPFL's one) is meant to be removed further down the road in development. I just keep it because it is very useful for testing things (manually) on the map and making sure others work. Possibly also useful for pitches, but useless for actual end users.